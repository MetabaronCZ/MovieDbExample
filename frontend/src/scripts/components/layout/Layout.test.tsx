import { getByText } from '@testing-library/dom';
import { describe, expect, it } from '@jest/globals';

import { Layout } from './Layout';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/layout/Layout', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(<Layout>CONTENT</Layout>, {
      wrapper: TestComponentWrapper,
    });
    expect(container).toMatchSnapshot();

    const content = getByText(container, 'CONTENT');
    expect(content).toBeInTheDocument();

    const main = container.getElementsByTagName('main');
    expect(main.length).toEqual(1);

    const header = container.getElementsByTagName('header');
    expect(header.length).toEqual(1);

    const footer = container.getElementsByTagName('footer');
    expect(footer.length).toEqual(1);
  });
});
