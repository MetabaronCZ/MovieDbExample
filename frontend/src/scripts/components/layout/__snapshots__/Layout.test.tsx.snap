// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layout/Layout should render properly 1`] = `
.c0 {
  display: flex;
  flex-direction: row;
  gap: 0px;
}

.c2 {
  display: flex;
  flex-direction: column;
  gap: 16px;
}

.c7 {
  display: flex;
  flex-direction: row;
  gap: 16px;
}

.c5 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 18px;
  line-height: 24px;
  font-weight: 400;
}

.c4 {
  padding: 0 16px;
}

.c6 {
  display: block;
  text-decoration: none;
  text-transform: uppercase;
  color: #98c3d9;
}

.c6:focus {
  outline: 1px dashed #002d56;
  outline-color: inherit;
}

.c13 {
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  color: inherit;
  text-decoration: underline;
}

.c13:hover {
  text-decoration: none;
}

.c13:focus {
  outline: 1px dashed #002d56;
  outline-color: inherit;
}

.c12 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  font-size: 12px;
  line-height: 24px;
  padding: 0 16px;
  color: #98c3d9;
  text-align: right;
}

.c14 {
  flex: 1;
  width: 100%;
  min-width: 0;
  padding: 16px 24px;
}

.c8 {
  flex: 1;
}

.c9 {
  list-style-type: none;
  display: flex;
  flex-direction: column;
  width: 100%;
}

.c10 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: block;
  padding: 8px 16px;
  color: #002d56;
  background: #98c3d9;
  text-decoration: none;
}

.c10:hover {
  background-color: #dae0e3;
}

.c10:focus {
  outline: 1px dashed #002d56;
  outline-color: #0af;
  outline-offset: -1px;
}

.c11 {
  font-family: Tahoma,Verdana,Arial,sans-serif;
  font-size: 14px;
  line-height: 24px;
  color: #002d56;
  display: block;
  padding: 8px 16px;
  color: #f2f7fa;
  background: #002d56;
  text-decoration: none;
}

.c11:hover {
  background-color: #0f426e;
}

.c11:focus {
  outline: 1px dashed #002d56;
  outline-color: #0af;
  outline-offset: -1px;
}

.c1 {
  width: 100%;
  min-width: 320px;
  min-height: 100vh;
  background: #f2f7fa;
}

.c3 {
  width: 256px;
  padding: 16px 0;
  background: #002d56;
}

<div>
  <div
    class="c0 c1"
  >
    <div
      class="c2 c3"
    >
      <header
        class="c4"
      >
        <h1
          class="c5"
        >
          <a
            class="c6"
            href="/"
          >
            Databáze filmů USA (1900 - 1999)
          </a>
        </h1>
      </header>
      <nav
        class="c7 c8"
      >
        <ul
          class="c9"
        >
          <li
            class=""
          >
            <a
              class="c10"
              data-discover="true"
              href="/"
            >
              Domů
            </a>
          </li>
          <li
            class=""
          >
            <a
              class="c11"
              data-discover="true"
              href="/movies"
            >
              Filmy
            </a>
          </li>
          <li
            class=""
          >
            <a
              class="c11"
              data-discover="true"
              href="/people"
            >
              Lidé
            </a>
          </li>
        </ul>
      </nav>
      <footer
        class="c12"
      >
        Copyright © 2025 
        <a
          class="c13"
          href="https://github.com/MetabaronCZ"
          target="_blank"
        >
          https://github.com/MetabaronCZ
        </a>
      </footer>
    </div>
    <main
      class="c2 c14"
    >
      CONTENT
    </main>
  </div>
</div>
`;
